metamodel/processor/src/main/java/org/hibernate/search/metamodel/processor/annotation/processing/impl/ProcessorVectorFieldProcessor.java

```diff
@@ -0,0 +1,143 @@
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingStep;
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingVectorFieldOptionsStep;
+import org.hibernate.search.metamodel.processor.model.BuiltInBridgeResolverTypes;
+
+public class ProcessorVectorFieldProcessor extends AbstractProcessorFieldAnnotationProcessor {
+	@Override
+	PropertyMappingFieldOptionsStep<?> initFieldMappingContext(PropertyMappingStep mappingContext, AnnotationMirror annotation,
+			String fieldName) {
+		int dimension = getAnnotationValueAsInt( annotation, "dimension", AnnotationDefaultValues.DEFAULT_DIMENSION );
+		PropertyMappingVectorFieldOptionsStep fieldContext = dimension == AnnotationDefaultValues.DEFAULT_DIMENSION
+				? mappingContext.vectorField( fieldName )
+				: mappingContext.vectorField( dimension, fieldName );
+
+		int m = getAnnotationValueAsInt( annotation, "m", AnnotationDefaultValues.DEFAULT_M );
+		if ( m != AnnotationDefaultValues.DEFAULT_M ) {
+			fieldContext.m( m );
+		}
+
+		int efConstruction =
+				getAnnotationValueAsInt( annotation, "efConstruction", AnnotationDefaultValues.DEFAULT_EF_CONSTRUCTION );
+		if ( efConstruction != AnnotationDefaultValues.DEFAULT_EF_CONSTRUCTION ) {
+			fieldContext.efConstruction( efConstruction );
+		}
+
+		VectorSimilarity vectorSimilarity = getVectorSimilarity( annotation );
+		if ( !VectorSimilarity.DEFAULT.equals( vectorSimilarity ) ) {
+			fieldContext.vectorSimilarity( vectorSimilarity );
+		}
+		Projectable projectable = getProjectable( annotation );
+		if ( !Projectable.DEFAULT.equals( projectable ) ) {
+			fieldContext.projectable( projectable );
+		}
+
+		Searchable searchable = getSearchable( annotation );
+		if ( !Searchable.DEFAULT.equals( searchable ) ) {
+			fieldContext.searchable( searchable );
+		}
+
+		String indexNullAs = getIndexNullAs( annotation );
+		if ( indexNullAs != null && !AnnotationDefaultValues.DO_NOT_INDEX_NULL.equals( indexNullAs ) ) {
+			fieldContext.indexNullAs( indexNullAs );
+		}
+
+		return fieldContext;
+	}
+
+	@Override
+	protected Optional<IndexFieldTypeFinalStep<?>> configureField(PropertyBindingContext bindingContext,
+			AnnotationMirror annotation,
+			ProcessorAnnotationProcessorContext context, Element element, TypeMirror fieldType) {
+		Class<?> vectorType;
+		Optional<Class<?>> loadableType = BuiltInBridgeResolverTypes.loadableType( fieldType );
+		if ( loadableType.isPresent() ) {
+			vectorType = loadableType.get();
+		}
+		else {
+			context.messager().printMessage( Diagnostic.Kind.ERROR, "Only float[]/byte[] fields are allowed.", element );
+			return Optional.empty();
+		}
+
+		VectorFieldTypeOptionsStep<?, ?> optionsStep = bindingContext.typeFactory().asVector( vectorType )
+				// it doesn't really matter since we won't be actually using this value anywhere,
+				// but since it's "optional" let's just keep it simple for now here:
+				.dimension( 10 );
```

Comment:
this is just an interesting thing to point out, as we don't really care about some things when building this model, we can skip reading them and just hardcode some values instead.

Reply:
Again fine as a temporary solution. Ultimately I'd hope we'd unify the processor implementations and just always read the value -- and trigger errors when invalid, such as `0` here.